src/Microsoft.Health.Fhir.Core/Features/Operations/BulkDelete/BulkDeleteOrchestratorJob.cs

```diff
@@ -0,0 +1,65 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Health.Fhir.Core.Features.Operations;
+using Microsoft.Health.Fhir.Core.Features.Operations.BulkDelete;
+using Microsoft.Health.Fhir.Core.Features.Search;
+using Microsoft.Health.JobManagement;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Api.Features.Operations.BulkDelete
+{
+    [JobTypeId((int)JobType.BulkDeleteOrchestrator)]
+    public class BulkDeleteOrchestratorJob : IJob
+    {
+        private IQueueClient _queueClient;
+        private ISearchService _searchService;
+
+        public BulkDeleteOrchestratorJob(
+            IQueueClient queueClient,
+            ISearchService searchService)
+        {
+            EnsureArg.IsNotNull(queueClient, nameof(queueClient));
+            EnsureArg.IsNotNull(searchService, nameof(searchService));
+
+            _queueClient = queueClient;
+            _searchService = searchService;
+        }
+
+        public async Task<string> ExecuteAsync(JobInfo jobInfo, IProgress<string> progress, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(jobInfo, nameof(jobInfo));
+            EnsureArg.IsNotNull(progress, nameof(progress));
+
+            BulkDeleteDefinition definition = JsonConvert.DeserializeObject<BulkDeleteDefinition>(jobInfo.Definition);
```

Comment:
Since this is so common, I was thinking we should have extensions for this like:
```csharp
var request = jobInfo.DeserializeDefinition<BulkDeleteDefinition>();
var jobResult = jobInfo.DeserializeResult<BulkDeleteResult>();
```

Reply:
It is hard to find all the paterns like this, but I think I got most of them replaced.